src/main/kotlin/se/uulm/snowballr/backend/validation/ProjectPaperValidator.kt

```diff
@@ -0,0 +1,43 @@
+package se.uulm.snowballr.backend.validation
+
+import arrow.core.EitherNel
+import arrow.core.raise.Raise
+import arrow.core.raise.either
+import arrow.core.raise.zipOrAccumulate
+import se.uulm.snowballr.backend.model.ValidationIssue
+import snowballr.ProjectOuterClass.Project.Paper as GrpcProjectPaper
+
+/**
+ * A validator for [GrpcProjectPaper] related requests.
+ */
+object ProjectPaperValidator {
+    fun validateGetRequest(request: GrpcProjectPaper.Get): EitherNel<ValidationIssue, Unit> = either {
+        zipOrAccumulate(
+            { ensureIdValidity("id", request.projectId) },
+            {
+                ensureRelativeProjectPaperIdValidity(request.relativeProjectPaperId)
+            },
+        ) { _, _ -> }
+    }
+
+    fun validateAddRequest(request: GrpcProjectPaper.Add): EitherNel<ValidationIssue, Unit> = either {
+        zipOrAccumulate(
+            { ensureIdValidity("id", request.projectId) },
+            { ensureIdValidity("id", request.paperId) },
```

Comment:
These are the incorrect field names.
```suggestion
            { ensureIdValidity("project_id", request.projectId) },
            { ensureIdValidity("paper_id", request.paperId) },
```